### ISSUE TITLE
Improve Flag activities to differentiate the object being reported vs the evidence for that report

### ISSUE THREAD

**Issue by ThisIsMissEm** [swicg/activitypub-trust-and-safety#3]
At present, the `Flag` activity contains a list of `objects` that are other activitystreams 2 objects that the flag is about, however, there's no differentiation of "this is evidence for the report" vs "this is a thing I want to report"

This leads to interesting compatibility bugs where for instance, as mastodon only understands reports about Actors, it requires both the actor and the note to be present in the Flag's `objects` list, and if software doesn't send that, the report fails to get created.

e.g., https://github.com/mastodon/mastodon/issues/28159

**Comment by trwnh** [comment]
preliminary idea, maybe there is or should be a semantic difference between `object` and `attachment`, as in, you might report the actor with various posts attached, or you might report the posts directly, or you might report both the actor and posts

this could also open the door to attaching evidence other than directly linking to the offending objects, you could attach screenshots for example?

**Comment by ThisIsMissEm** [comment]
> You could also report an instance, where multiple actors contribute to the reason.

@jfinkhaeuser you could, but this would be unlikely to be a Flag activity (who would you be sending it to?)

unless we extend how Flag's are used to not just be for reporting content/conduct, but also for communicating to a non-participant. Normally a Flag is "this specific content or actor is doing something bad, send a Flag activity to their instance actor to let their moderators know" or same thing but "send this Flag to other servers involved in the conversation to let them know of an evolving situation"

ultimately, a server receiving a Flag activity needs to know "do I care about the objects within to create a report about this?"

**Comment by jfinkhaeuser** [comment]
Yes, I think I joined this conversation from a slightly different angle. It would/should be distinct from Flag, but possibly have overlap. I added more on #14, and think you should disregard my previous comment here - sorry!
